person.rb

```diff
@@ -13,9 +15,43 @@ def of_age?
     (@age.is_a? Numeric) && (@age >= 18)
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def can_use_services?
     @age >= 18 || @parent_permission
   end
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? "#{@nameable.correct_name.strip[0..9]}..." : @nameable.correct_name
+  end
+end
+
+person = Person.new('maximilianus')
+person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+puts capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
+puts capitalized_trimmed_person.correct_name
```

Comment:
**[OPTIONAL]**

- Hey :wave: , To improve your code organization, all these classes should be in a separate file, this will make your code more readable, maintainable and more professional.

Reply:
Thanks for the advice